analytical_engine/java/grape-jdk/src/main/java/com/alibaba/graphscope/fragment/SimpleFragment.java

```diff
@@ -16,6 +17,14 @@
      * @return underlying fragment type.
      */
     String fragmentType();
+
```

Comment:
reconsider this fragment, do we really need a new one?
is this fragment different from the interface of `ImmutableEdgecutFragment` or `EdgecutFragment` by design?

Reply:
I think a new fragment interface is necessary, the reasons are:
- There are indeed some minior difference(mainly in `AdjList`) between `ImmutableEdgecutFragment` and `ArrowProjectedFragment`. We need adaptors to makes them looks same.
- In C++, as long as `ArrowProjected` and `ImmutableEdgecutFragment` has same method signature, they can be used interchangeably. But not in Java. We need common base class for them in java.

Reply:
OK. how about we use `IFragment` to avoid confusing? where the `I` represents Interface.

Reply:
OK, No prob.